packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -459,41 +472,29 @@ function WorklistDataGrid(props) {
     setFocusedRow(selectionDetails.rowIndex);
     setFocusedCol(selectionDetails.columnIndex);
 
-    const rangeInfo = selectionDetails.selectedByKeyboard ? rowSelection.current.rangeUsingLastIndividuallySelectedRow : rowSelection?.current?.rangeUsingFocusedRow;
-    if (!selectionDetails.multiSelect) {
-      if (hasSelectableRows) {
-        // regular click or space key
-        selectRow(selectionDetails.rowId, selectionDetails.rowIndex);
-      }
-    } else if (hasSelectableRows) {
-      // Shift key is pressed
-      if (rangeInfo?.anchorRow) {
-        // ACTION: select multiple rows based on anchor
-        selectMultipleRows(selectionDetails.rowIndex, rangeInfo);
-        rangeInfo.previousSelectionEndRow = selectionDetails.rowIndex;
-      } else {
-        // no anchor row
-        selectRow(selectionDetails.rowId, selectionDetails.rowIndex);
+    if (!hasSelectableRows) {
+      if (selectionDetails.isShiftKeyPressed) {
+        // Shift+Space, Shift+Click, Shift+Up, Shift+Down
+        isRowSelectionModeToggledByGrid.current = !hasSelectableRows;
+        multiSelectRange.current = { start: selectionDetails.rowIndex, end: null };
+        selectMultipleRows(multiSelectRange.current.start, multiSelectRange.current.end, selectionDetails.rowIndex);
```

Comment:
multiSelectRange is a ref hook.  Why do you need to pass its values to the selectMultipleRows function.  It seems like that function could obtain the values directly.

Reply:
Changed